python/sglang/srt/distributed/device_communicators/custom_all_reduce.py

```diff
@@ -394,7 +394,8 @@ def custom_all_reduce(self, input: torch.Tensor) -> Optional[torch.Tensor]:
                 if _is_hip:
                     return self.all_reduce_reg(input)
                 else:
-                    return self.all_reduce(input, registered=True)
+                    registered = not get_bool_env_var("SGLANG_MEMORY_SAVER_CUDA_GRAPH")
```

Comment:
nit

1. maybe put it in constructor (`def __init__: self.blahblah = get env var`) to avoid calling get env var overhead in runtime
2. maybe use the new envs.py approach to handle env var